src/plugins/gui/pluginUtils.ts

```diff
@@ -128,3 +128,16 @@ export const debugSpewStore = async (store: EdgeDataStore): Promise<void> => {
   console.log('*** theStore ***')
   console.log(JSON.stringify(theStore, null, 2))
 }
+
+export const assert = (condition: boolean, message: string) => {
+  if (!condition) throw new Error(message)
+}
```

Comment:
Might not be worth it, but we can also have a generic flavor:

```suggestion
export const assert = (
  condition: boolean,
  ErrorType: new (...args: any[]) => Error = Error,
  props: Record<string, any> = { message: 'An error occurred' }
) => {
  if (!condition) {
    const error = new ErrorType()
    Object.assign(error, props)
    throw error
  }
}

// usage:
assert(true, undefined, { message: "Something went wrong" })
assert(minMatch != null, FiatProviderError, { providerId, errorType: 'underLimit', errorAmount: Number(minMatch[1]), displayCurrencyCode: minMatch[2] })

```

Reply:
I like it, but I don't want to add the code now as this PR won't be using it. I'm open to it being added once it once it's used.